src/main/kotlin/com/tschuchort/compiletesting/KotlinCompilation.kt

```diff
@@ -475,8 +489,7 @@ class KotlinCompilation {
 
 		val k2JvmArgs = commonK2JVMArgs().also {
 			it.freeArgs = sourcePaths
-			it.pluginClasspaths = (it.pluginClasspaths?.toList() ?: emptyList<String>() + getResourcesPath())
-					.distinct().toTypedArray()
+			it.pluginClasspaths = it.pluginClasspaths?.plus(arrayOf(getResourcesPath()))
```

Comment:
This will still be `null` if the `it.pluginClassPaths` is initially null. You need something like `(it.pluginClassPaths ?: emptyArray()) + arrayOf(getResourcesPath())` or similar. Maybe put a `.distinct()` in there as well just to be sure, although it's probably not necessary.

Reply:
Oh, sorry, it's true. I was thinking about `pluginClasspaths` which is initialized with an empty list. I'll fix tomorrow morning, thank you!!

Reply:
It's a pleasure to "work" with you!!

Reply:
Fixed!

Reply:
Thanks, it means a lot to me to hear that. Hopefully I will find some time to pay it forward and do some contributions to an ArrowKt project.